Composed/Core/DataSources/EmptyDataSource.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+open class EmptyDataSource: DataSource {
+    public var isEmpty: Bool { return true }
+    public weak var updateDelegate: DataSourceUpdateDelegate?
+    public var numberOfSections: Int { return 1 }
```

Comment:
By default I would expect "empty" to mean literally nothing, including zero sections. The main use for this is for headers/footers, or to "fill in" areas that expect a data source, such as a `SegmentedDataSource`?

If it's for headers/footers I wonder if a more specific implementation would be useful? If it's for `SegmentedDataSource` I think this could be zero?

Reply:
So, Segmented no longer requires this. You can just do `segmented.setSelectedIndex(nil, animated: false)`

Reply:
It is mostly for headers/footers but I think you're right that this isn't the best use for it. I think I'll pull. this out since its unused otherwise. When I added tests for it today I realised it was kinda pointless since it can never had any other values anyway, not without UI.

Reply:
1003a0ca297f2f47664856461916166a3ee6565d